src/coreclr/jit/emitarm64.cpp

```diff
@@ -18577,6 +18629,69 @@ void emitter::emitIns_Call(EmitCallType          callType,
     return encoding;
 }
 
+/*****************************************************************************
+ *
+ *  Returns the encoding for the field 'tszh:tszl:imm3' at bit locations '23-22:20-19:18-16'.
+ */
+
+/*static*/ emitter::code_t emitter::insEncodeSveElemsizeWithShift_tszh_tszl_imm3(const insOpts opt,
+                                                                                 ssize_t       imm,
```

Comment:
Shift immediates are notoriously awkward :)

Does any of this code overlap with the existing `insEncodeShiftImmediate()`?
You can see how it's used in `insEncodeSveShift_23_to_22_9_to_0()`
`insEncodeShiftImmediate()` is also used by NEON, as shift immediates tend to be fairly standard.

Reply:
I might be able to use `insEncodeShiftImmediate` to tweak the `imm` instead of doing it manually.